python/ray/ml/utils/check_ingest.py

```diff
@@ -75,7 +78,16 @@ def train_loop_per_worker():
                     batch_delay = time.perf_counter() - batch_start
                     batch_delays.append(batch_delay)
                     num_batches += 1
-                    num_bytes += int(batch.memory_usage(index=True, deep=True).sum())
+                    if isinstance(batch, pd.DataFrame):
+                        num_bytes += int(
+                            batch.memory_usage(index=True, deep=True).sum()
+                        )
+                    elif isinstance(batch, np.ndarray):
+                        num_bytes += batch.nbytes
+                    else:
+                        # NOTE: This isn't recursive and will just return the size of
```

Comment:
seems this is the recommend recursive way: https://code.activestate.com/recipes/577504/: but I think we can leave it as a TODO.

Reply:
Yeah I was going to open up a separate issue for this, since we currently calculate the byte size of simple blocks with this top-level `sys.getsizeof()` but we might want to use a recursive recipe both there and here.

Reply:
In the future we can use BlockAccessor.for_block(b).size_bytes()

Reply:
@ericl For sure, but it's batch, not a block! So
```python
block = BlockAccessor.batch_to_block(batch)
num_bytes += BlockAccessor.for_block(block).nbytes
```

Reply:
Got it.